src/types/index.ts

```diff
@@ -345,18 +365,20 @@ export class StarknetContractFactory {
     public abiPath: string;
     private constructorAbi: starknet.CairoFunction;
     public metadataPath: string;
+    public casmPath: string;
     private classHash: string;
 
     constructor(config: StarknetContractFactoryConfig) {
         this.hre = config.hre;
         this.abiPath = config.abiPath;
         this.abi = readAbi(this.abiPath);
         this.metadataPath = config.metadataPath;
+        if (config.casmPath)this.casmPath = config.casmPath;
 
         // find constructor
         for (const abiEntryName in this.abi) {
             const abiEntry = this.abi[abiEntryName];
-            if (abiEntry.type === "constructor") {
+            if (abiEntry.type === "constructor" || abiEntry.name === "constructor") {
```

Comment:
https://github.com/Shard-Labs/starknet-hardhat-plugin/pull/328/commits/cb381b88cfc98098f854b505296b7e67d32becbc

Reply:
How about an early exit here? I.e. after the constructor is found, the loop can be broken out of.

Reply:
Moved that check on this function. It exits before the loop begins. https://github.com/Shard-Labs/starknet-hardhat-plugin/blob/cb381b88cfc98098f854b505296b7e67d32becbc/src/utils.ts#L383

Reply:
I was referring to the loop which is still present: `for (const abiEntryName in this.abi)`
I meant that its body can look like this
```ts
            if (isEntryAContructor(abiEntry, config.hre.config.paths, this.casmPath)) {
                this.constructorAbi = <starknet.CairoFunction>abiEntry;
                break; // the early exit I had in mind
            }
```

But this can also be addressed as a separate issue/PR.